packages/spacecat-shared-rum-api-client/src/functions/variant.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const VARIANT_CHECKPOINT = 'variant';
+
+function getOrCreateLanguageObject(languageInsights, language) {
+  let languageObject = languageInsights.find((l) => l.language === language);
+  if (!languageObject) {
+    languageObject = {
+      language,
+      count: 0, // Tracks the total number of events for this page language
+      mismatches: {
+        type1: {
+          preferredLanguages: {}, // Type 1 mismatches
+        },
+        type2: {
+          preferredLanguages: {}, // Type 2 mismatches
+        },
+        type3: {
+          preferredLanguages: {}, // Type 3 mismatches
+        },
+      },
+      regions: {}, // Tracks the count of events for each region
+    };
+    languageInsights.push(languageObject);
+  }
+  return languageObject;
+}
+
+function handler(bundles) {
+  const languageInsights = [];
+
+  for (const bundle of bundles) {
+    let preferredLanguages = [];
+    let pageLanguage = null;
+    let userRegion = null;
+
+    for (const event of bundle.events) {
+      if (event.checkpoint === VARIANT_CHECKPOINT) {
+        const { target, source } = event;
+
+        if (source === 'preferred-languages') {
+          preferredLanguages = target.split(',').map((lang) => lang.trim());
+        }
+
+        if (source === 'page-language') {
+          pageLanguage = target;
+        }
+
+        if (source === 'user-region') {
+          userRegion = target;
+        }
```

Comment:
i'd suggest an else-if here to prevent other conditions from being evaluated:
```suggestion
        if (source === 'preferred-languages') {
          preferredLanguages = target.split(',').map((lang) => lang.trim());
        } else if (source === 'page-language') {
          pageLanguage = target;
        } else if (source === 'user-region') {
          userRegion = target;
        }
```